pkgs/test/top-level/stage.nix

```
@@ -0,0 +1,116 @@
+# run like this:
+#  nix-build pkgs/test/top-level/stage.nix
+{
+  localSystem ? {
+    system = builtins.currentSystem;
+  },
+}:
+
+with import ../../top-level { inherit localSystem; };
+
+let
+  # To silence platform specific evaluation errors
+  discardEvaluationErrors =
+    e:
+    let
+      res = builtins.tryEval e;
+    in
+    res.success == res.value;
```

```suggestion
  discardEvaluationErrors = e: (builtins.tryEval e).success -> e;
```

If I am not mistaken the two are equivalent for booleans, but the second version is more easily understood and will throw an error if e is not a boolean.